Assets/UnityIO.cs

```diff
@@ -0,0 +1,48 @@
+using UnityEngine;
+using System;
+using System.IO;
+using System.Collections;
+
+public class UnityIO
+{
+	public class Response {
+		public string error;
+		public byte[] data;
+
+		public Response(string e, byte[] d) {
+			error = e;
+			data = d;
+		}
+		public Response(string e, string d) {
+			error = e;
+			data = Convert.FromBase64String(d);
+		}
+	}
+
+	public delegate void IORequestCallback(object userData, Response response);
+
+	public IEnumerator FetchData(Uri uri, IORequestCallback callback, object userData) {
+
+		bool networkRequest = false;
+		Response response;
+		if (uri.Scheme == Uri.UriSchemeHttp || uri.Scheme == Uri.UriSchemeHttps) {
+			networkRequest = true;
+		}
+
+		if (networkRequest) {
+			WWW request = new WWW(uri.AbsoluteUri);
+			yield return request;
+			response = new Response(request.error, request.bytes);
+		} else {
+			yield return 0;
+			var obj = Resources.Load(uri.AbsolutePath);
+			if (obj == null) {
+				response = new Response("Resource not found", "");
+			} else {
+				response = new Response(null, obj.ToString());
```

Comment:
This doesn't work because `obj.ToString()` returns the name of the file.

`Resources.Load` only load asset that are Unity assets, basically in a case similar to this:

```c#
// If tileAsset is a .prefab
GameObject go = Instantiate (Resources.Load("tileAsset")) as GameObject;
```
That we can use for loading materials/textures etc.. 

For text files (for example a `.json` resource) you can simply load it like that:

```c#
TextAsset tileTextAsset = Resources.Load("mapzen-tile.json") as TextAsset;
```

We could add an interface to load Unity assets, using the proper way to load them (similar to web requests):

```c#
// 1. Construct the URI considering using Application.dataPath + "/Resources/" as base path
// 2. Load the resource async
ResourceRequest request = Resources.LoadAsync(uri.aboslutePath);
yield return request;
// 3. Construct the response
```

If we want to stick with using `byte[]` we can load file as a binary in a synchronous way like this https://github.com/tangrams/tangram-unity/commit/7c882b383f7a0ee86fe8e28d82d8517cd104abf7, feel free to cherry-pick the commit. I think it will be enough for now, we can add the interface for unity assets when we start dealing with more resources.